src/base/http/requesthandler.h

```diff
@@ -28,16 +28,21 @@
 
 #pragma once
 
+#include "iconnectionhandler.h"
+
 namespace Http
 {
     struct Environment;
     struct Request;
     struct Response;
 
-    class IRequestHandler
+    class RequestHandler : public IConnectionHandler
```

Comment:
Seems only RequestHandler inherits IConnectionHandler. Do you intend to implement other inheritors? If not, then maybe there is no point in extracting this interface, but merge it into  RequestHandler instead?

Reply:
> Seems only RequestHandler inherits IConnectionHandler. Do you intend to implement other inheritors? If not, then maybe there is no point in extracting this interface, but merge it into RequestHandler instead?

Yes, #12542 requires to feed the content in form of chunks in torrent piece length asynchronously. The linear and limiting interface for RequestHandler wouldn't allow it.